core/src/test/scala/org/apache/spark/deploy/history/FsHistoryProviderSuite.scala

```diff
@@ -1705,6 +1705,45 @@ abstract class FsHistoryProviderSuite extends SparkFunSuite with Matchers with P
     provider.stop()
   }
 
+  test("SPARK-41447: clean up expired event log files that don't exist in listing db") {
```

Comment:
@shuyouZZ This test case seems to pass without your code. Could you fix it to be a valid test case which fails without your patch?

Reply:
> @shuyouZZ This test case seems to pass without your code. Could you fix it to be a valid test case which fails without your patch?

Checked the previous code, i will add an extra class TestFsHistoryProvider in test to validate the patch.